### ISSUE TITLE
Improve support for ESM

### ISSUE THREAD

**Issue by citycide** [d-fischer/twitch#122]
## Feature Request

Node 14 is released and will soon become LTS, so ESM is pretty usable out of the box in Node. I'm experimenting with converting my app to use modules (without TypeScript), and the UX for the twitch libraries could be improved.

https://github.com/d-fischer/twitch/blob/9fb614946c95bd14820cfb61f0e353b06bb9fa8b/packages/twitch/src/index.ts#L4

As is, intellisense calls this default import a class (because of the type definitions) but really, [to Node](https://nodejs.org/api/esm.html#esm_import_statements), it's the common JS `module.exports` object.

```js
import TwitchClient from "twitch"
```

So to make it work in a Node ES module we would have to use its `default` property to get the class itself:

```js
import twitch from "twitch"
const TwitchClient = twitch.default
```

Unfortunately the type definitions still don't allow intellisense to pick this up. TypeScript has some interop features for this but standard Node does not. Most of these issues could be sidestepped by avoiding default exports (they're a JS language design mistake in my opinion), and only using named exports.

For backward compatibility a named export could be introduced alongside the current default export, which would at least alleviate these problems today:

```js
export { TwitchClient };
export default TwitchClient;
```

```js
import twitch from "twitch"
const { TwitchClient } = twitch
```

This would also apply to the other packages like the chat client, and other exposed default exports.

An ES module build could also be introduced but that's a more involved proposition.

**Comment by d-fischer** [comment]
This is currently in the works, there will be a separate ESM build in the package in version 4.1 - keep an eye out, I will publish a prerelease version soon™ 🙂

**Comment by d-fischer** [comment]
Currently blocked by lukeed/bundt#3.

**Comment by d-fischer** [comment]
I published `4.1.0-pre.0` anyway, but WebHooks are broken in the CJS version for now.

**Comment by d-fischer** [comment]
In the latest prerelease, WebHooks are not broken anymore. I'm closing this issue now that things are working.
